pyscf/tdscf/uks_sf.py

```diff
@@ -0,0 +1,220 @@
+#!/usr/bin/env python
+# Copyright 2014-2020 The PySCF Developers. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy
+from pyscf import symm
+from pyscf import lib
+from pyscf.lib import logger
+from pyscf.tdscf import uhf
+from pyscf.scf import uhf_symm
+from pyscf.data import nist
+from pyscf.dft.rks import KohnShamDFT
+from pyscf import __config__
+
+
+class TDA_SF(uhf.TDA_SF):
+    def nuc_grad_method(self):
+        from pyscf.grad import tduks
+        return tduks.Gradients(self)
+
+class TDDFT_SF(uhf.TDA_SF):
+    pass
+    # print('Remember to set collinear_samples in SF-TDDFT, \
+    #        the default value is 200.')
+    # def nuc_grad_method(self):
+    #     from pyscf.grad import tduks
+    #     return tduks.Gradients(self)
+    
+TDUKS_SF = TDDFT_SF
+
+class CasidaTDDFT(TDDFT_SF, TDA_SF):
+    '''Solve the Casida TDDFT formula
+       [ A  B][X]
+       [-B -A][Y]
+    '''
+    
+    init_guess = TDA_SF.init_guess
+
+    def gen_vind(self, mf=None,extype=0,collinear_samples=200):
```

Comment:
Is the `collinear_samples` option necessary?

Reply:
The "collinear_samples" option is used to control the number of one-dimensional distribution points in spin-flip TDDFT/ TDA. It is a same name with the number of collinear samples points for strongly spin-polarized places in a multicollinear GKS object.